### PR TITLE
Be vintageous friendly

### PR DESCRIPTION
**Pull request by mreq**
continued from  #372

### CODE REVIEW

**Comment by mreq** [comment]
Somehow this didn't work when I first tried it. It does, however, work now and is definitely a better approach.

`{ "key": "setting.command_mode", "operator": "equal", "operand": false }` is enough

File: core/commands/commit.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def run_async(self, repo_path=None, include_unstaged=False, amend=False):
         else:
             view.set_syntax_file("Packages/GitSavvy/syntax/make_commit.tmLanguage")
 
+        if savvy_settings.get("vintageous_friendly", False) is True:
```
Comment (divmain) [inline core/commands/commit.py]: Can you add a comment block above this line, explaining why it is here and what it does?  Add the same for `log_graph.py` and `ui.py`.

**Comment by divmain** [comment]
Awesome, I like this much better.  Once you add the comments I requested, I'll merge this in!  I may revisit at some point to see if this can be further abstracted, so that Vintageous-specific settings become unnecessary.  But this seems a reasonable step for now.  Thanks for your work on this!

**Comment by mreq** [comment]
I've created a GsHandleVintageousCommand which has it's own docstring and keeps the code DRY. Hopefully placed it right.

**Comment by divmain** [comment]
That's perfect.  Thanks again!

**Comment by asfaltboy** [comment]
This PR actually causes a bit of pain for me. Since the default mode when opening a view is command mode, to use GitSavvy shortcuts I must change to insert. Even changing between dashboards using TAB requires changing to insert mode every time. Honestly, even though the previous mode was lacking some vim-friendliness , at least it required much less keystrokes.

**Comment by mreq** [comment]
@asfaltboy set the `vintageous_enter_insert_mode` option to `true`

**Comment by asfaltboy** [comment]
Brilliant! Why won't we set [these to True by default][1]?

[1]: https://github.com/divmain/GitSavvy/blob/master/common/commands/view_manipulation.py#L68-L70

**Comment by mreq** [comment]
Indifferent to me. Might be a good idea, as it mimics the default non-vintageous behaviour. What do you think, @divmain?

**Comment by asfaltboy** [comment]
We should definitely document this feature if possible

**Comment by mreq** [comment]
Well, it is explained in the [default settings](https://github.com/divmain/GitSavvy/blob/master/GitSavvy.sublime-settings#L258) file.

**Comment by asfaltboy** [comment]
Yes, this is why I just opened issue #430 .

**Pull request merged**